source/Mission.cpp

```diff
@@ -1248,15 +1264,27 @@ Mission Mission::Instantiate(const PlayerInfo &player, const shared_ptr<Ship> &b
 	while(!destinations.empty())
 	{
 		// Find the closest destination to this location.
-		DistanceMap distance(path);
+		DistanceMap distance(path, wormholeStrategy, requiresJumpDrive);
 		auto it = destinations.begin();
 		auto bestIt = it;
+		int bestDays = distance.Days(*bestIt);
+		if(bestDays < 0)
+			bestDays = numeric_limits<int>::max();
 		for(++it; it != destinations.end(); ++it)
-			if(distance.Days(*it) < distance.Days(*bestIt))
+		{
+			int days = distance.Days(*it);
+			if(days < 0)
+				days = numeric_limits<int>::max();
+			if(days < bestDays)
+			{
 				bestIt = it;
+				bestDays = days;
+			}
+		}
 
 		path = *bestIt;
-		jumps += distance.Days(*bestIt);
+		// If currently unreachable, this system does not add to the deadline.
+		jumps += max(0, distance.Days(*bestIt));
 		destinations.erase(bestIt);
 	}
 	DistanceMap distance(path);
```

Comment:
Yes, I think you're right.
Relating to #5085, should these options also be added for payments (should payments default to using the deadline calculation route jump count?) and LocationFilter (that would need to default to WormholeStrategy::NONE, because of what you mentioned earlier, regardless of the deadlineOption)?
If so, should those go in a separate PR?